graph/import.go

```diff
@@ -19,21 +24,40 @@ func (s *TagStore) CmdImport(job *engine.Job) engine.Status {
 		return job.Errorf("Usage: %s SRC REPO [TAG]", job.Name)
 	}
 	var (
-		src          = job.Args[0]
-		repo         = job.Args[1]
-		tag          string
-		sf           = utils.NewStreamFormatter(job.GetenvBool("json"))
-		archive      archive.ArchiveReader
-		resp         *http.Response
-		stdoutBuffer = bytes.NewBuffer(nil)
-		newConfig    runconfig.Config
+		src           = job.Args[0]
+		repo          = job.Args[1]
+		tag           string
+		sf            = utils.NewStreamFormatter(job.GetenvBool("json"))
+		archiveReader archive.ArchiveReader
+		resp          *http.Response
+		stdoutBuffer  = bytes.NewBuffer(nil)
+		newConfig     runconfig.Config
 	)
 	if len(job.Args) > 2 {
 		tag = job.Args[2]
 	}
 
 	if src == "-" {
-		archive = job.Stdin
+		buf, _ := ioutil.ReadAll(job.Stdin)
```

Comment:
Already thought about this, I'm looking for suggestions because the reader is consumed when inside untar